lib/remoteproc/elf_loader.c

```diff
@@ -650,17 +648,17 @@ metal_phys_addr_t elf_get_entry(void *elf_info)
 		return METAL_BAD_PHYS;
 
 	if (elf_is_64(elf_info) == 0) {
-		Elf32_Ehdr *elf_ehdr = (Elf32_Ehdr *)elf_info;
+		Elf32_Ehdr *elf_ehdr = elf_info;
 		Elf32_Addr e_entry;
 
 		e_entry = elf_ehdr->e_entry;
-		return (metal_phys_addr_t)e_entry;
+		return e_entry;
```

Comment:
should still use the metal_phys_addr_t to force casting, to avoid compilation warning in some system as the size of the entry may not be the same as metal_phys_addr_t

Reply:
Fix